Bug 1907628: OpenStack Documentation for MachineSets with Multiple Networks (#4463)
openshift/installer: docs/user/openstack/README.md

Groups of Compute nodes are managed using the [MachineSet][machine-set-code] resource. It is possible to create additional MachineSets post-install, for example to assign workloads to specific machines.

> When running on OpenStack, the MachineSet has platform-specific fields under `spec.template.spec.providerSpec.value`. More details about what values can be set in the `providerSpec` can be found in the [API definition](provider-spec-definition).

Reviewer: ```suggestion
When running on OpenStack, the MachineSet has platform-specific fields under `spec.template.spec.providerSpec.value`. For more information about the values that you can set in `providerSpec`, see [the API definition](provider-spec-definition).
```

          availabilityZone: <optional_openstack_availability_zone>
```

[provider-spec-definition]: https://github.com/openshift/cluster-api-provider-openstack/blob/155384b859c5b2fb5b7f11c9111d3f8e4f3066bd/pkg/apis/openstackproviderconfig/v1alpha1/types.go#L31

> #### Defining a New MachineSet With Multiple Networks

Reviewer: ```suggestion
#### Defining a New MachineSet That Uses Multiple Networks
```

> When attaching multiple networks to a machineset, the primarySubnet in the `providerSpec` need to be set to the `machines subnet` or the node will be unable to join the cluster. The primary subnet is where a nodes kuberentes endpoints are published. Once this is properly set, you can should add all of the networks that you want attached to the machines in the `Networks` list in the `providerSpec`. Note that the network that the primary subnet is part of should also be added to this list.

Reviewer: ```suggestion
To attach multiple networks to a machine set, the `primarySubnet` value in `providerSpec` must be set to the `machines subnet`. If the value is not set, the node cannot join the cluster. 

Nodes' Kubernetes endpoints are published on the primary subnet. After you set the subnet, add all of the networks that you want to attach to your machines to the `Networks` list in `providerSpec`. You must also add the network that the primary subnet is part of to this list. 
```

Not 100% about what you mean by `machines subnet` here.

Author: I will update with a link to the definition. Chances are someone else will also be confused by that

Reviewer: Ah! `machines subnet` vs. `machinesSubnet`.  👍

          availabilityZone: <optional_openstack_availability_zone>
```

[provider-spec-definition]: https://github.com/openshift/cluster-api-provider-openstack/blob/155384b859c5b2fb5b7f11c9111d3f8e4f3066bd/pkg/apis/openstackproviderconfig/v1alpha1/types.go#L31

> #### Defining a New MachineSet That Uses Multiple Networks

Reviewer 2: do you need the word "New"? Defining usually means making the first of something.

> To attach multiple networks to a machineSet, the `primarySubnet` value in the `providerSpec` must be set to the [machines subnet](./customization.md#cluster-scoped-properties). If this value is not set correctly, the nodes will be unable to join the cluster.

Reviewer 2: here you use "machineSet" (lower case), but in L447 you use "MachineSet" is that correct?

Reviewer 2: You mention `primarySubnet` in `providerSpec` in ./customization.md#cluster-scoped-properties. 
and it sounds as if you are saying that the user needs to set `primarySubnet` somewhere, but if i am reading this correctly you mean that the user needs to set the `machineSubnet` which in turn will set the `primarySubent` in the `providerspec` of the generated machineSet... right?  anyways. I think its a bit hard to follow.

Author: I tried cleaning up the sentence. Unfortunately, I cant clean up the UX. I agree that its unclear and I dont like it. Lets make an epic for that if we want to change it.

> To attach multiple networks to a MachineSet, the `primarySubnet` value in the `providerSpec` must be the same OpenStack subnet as the [machines subnet](./customization.md#cluster-scoped-properties) in the `install-config.yaml`. If this value is not set correctly, the nodes will be unable to join the cluster.

Reviewer: Not set "correctly," or not set so that it contains `primarySubnet`? 

Just "correctly" could imply that there are a variety of ways to screw up the value, with what's in the previous sentence being just one of them. 

(Could expand if that's actually the case.)

Author: When defining a MachineSet with multiple networks, the `primarySubnet` value in the `providerSpec` must be set to the OpenStack subnet that you want the Kubernetes endpoints of nodes to be published on. For most use cases, this is the same subnet as the [machinesSubnet](./customization.md#cluster-scoped-properties) in the `install-config.yaml`.

How is this? I think it more accurately reflects the usage

Reviewer: Works for me. :+1:

Author: Take a look at what I pushed and see if it reads more clearly to you

> Nodes' Kubernetes endpoints are published on the primary subnet. After you set the subnet, add all of the networks that you want to attach to your machines to the `Networks` list in `providerSpec`. You must also add the network that the primary subnet is part of to this list.

Reviewer 3: Is ```Nodes'``` added on purpose?

Reviewer: "The Kubernetes endpoints of nodes are published..." ?

Author: Yeah, The node's kubernetes endpoints are being published on the primary subnet
